### ISSUE TITLE
Use streaming JSON parser for combining JSON results

### ISSUE THREAD

**Issue by chrisdone** [tirumaraiselvan/graphql-engine#21]
Currently we're using aeson's `decode` to parse into `Value` structures, lookup the `data` field, then union the `HashMap`s, which is convenient but less efficient than if we parsed the boundaries of the object using the attoparsec combinators, located the `data` key boundaries, and then just concatenated the bytestrings together at the right point as a `Builder` (the `EncJSON` uses this). [Here](https://github.com/chrisdone/streaming-parsers/blob/master/parsax/src/Data/Parsax/Json.hs#L179-L239) is an example of parsing directly.

**Comment by jberryman** [comment]
I just feel icky doing performance work when we don't have benchmarks. It's first of all hard to tell if the effort is worth it, and second we don't get to feel good about any big gains we've attained.

@tirumaraiselvan do you have any thoughts on benchmarking as it relates to this feature?

**Comment by jberryman** [comment]
I can take this one.

**Comment by tirumaraiselvan** [comment]
Yeah, these are optimizations that we better do more systematically. So let's put this on hold.
